docs/design.md

```diff
@@ -1,37 +1,143 @@
-# OpenTelemetry Service Design Specs
+# OpenTelemetry Service Architecture
 
 This document describes the architecture design and implementation of
-OpenTelemetry Agent and OpenTelemetry Collector.
-
-# Table of contents
-- [Goals](#goals)
-- [OpenTelemetry Agent](#opentelemetry-agent)
-    - [Architecture overview](#agent-architecture-overview)
-    - [Communication](#agent-communication)
-    - [Protocol Workflow](#agent-protocol-workflow)
-    - [Implementation details of Agent server](#agent-implementation-details-of-agent-server)
-        - [Receivers](#agent-impl-receivers)
-        - [Agent Core](#agent-impl-agent-core)
-        - [Exporters](#agent-impl-exporters)
-- [OpenTelemetry Collector](#opentelemetry-collector)
-    - [Architecture overview](#collector-architecture-overview)
-
-## <a name="goals"></a>Goals
-
-* Allow enabling/configuring of exporters lazily. After deploying code,
-optionally run a daemon on the host and it will read the collected data and
-upload to the configured backend.
-* Binaries can be instrumented without thinking about the exporting story.
-Allows open source binary projects (e.g. web servers like Caddy or Istio Mixer)
-to adopt OpenTelemetry without having to link any exporters into their binary.
-* Easier to scale the exporter development. Not every language has to
-implement support for each backend.
-* Custom daemons containing only the required exporters compiled in can be
-created.
-
-## <a name="opentelemetry-agent"></a>OpenTelemetry Agent
-
-### <a name="agent-architecture-overview"></a>Architecture Overview
+OpenTelemetry Service.
+
+## Summary
+
+OpenTelemetry Service is an executable that allows to receive telemetry data, optionally transform it and send the data further.
+
+The Service supports several popular open-source protocols for telemetry data receiving and sending as well as offering a pluggable architecture for adding more protocols.
+
+Data receiving, transformation and sending is done using Pipelines. The Service can be configured to have one or more Pipelines. Each Pipeline includes a set of Receivers that receive the data, a series of optional Processors that get the data from receivers and transform it and a set of Exporters which get the data from the Processors and send it further outside the Service. The same receiver can feed data to multiple Pipelines and multiple pipelines can feed data into the same Exporter.
+
+## Pipelines
+
+Pipeline defines a path the data follows in the Service starting from reception, then further processing or modification and finally exiting the Service via exporters.
+
+Pipelines can operate on 2 telemetry data types: traces and metrics. The data type is a property of the pipeline defined by its configuration. Receivers, exporters and processors used in a pipeline must support the particular data type otherwise ErrDataTypeIsNotSupported will be reported when the configuration is loaded. A pipeline can be depicted the following way:
```

Comment:
nit: use `ErrDataTypeIsNotSupported` since it is an actual type in code.

Reply:
I think I already have `ErrDataTypeIsNotSupported` in the text. Maybe I miss something?

Reply:
I think I understand, you mean the formatting with backticks. Adding.

Reply:
Fixed.